### PR TITLE
Add idb check, remove ptyhon version check for dump

### PR DESCRIPTION
**Pull request by KazuCocoa**
- Remove Python 2 check since we no longer need it https://github.com/appium/appium/pull/12609
    - https://github.com/appium/appium-doctor/issues/72
- Replace fbsctl with idb
    - https://github.com/appium/appium-ios-simulator/pull/224

### CODE REVIEW

File: test/general-specs.js [hunk 1]
```diff
@@ -195,55 +195,4 @@ describe('general', function () {
         equal('mjpeg-consumer module is required to use MJPEG-over-HTTP features. Please install it with `npm i -g mjpeg-consumer`.');
     });
   }));
-
-  describe('OptionalPythonVersionCheck', withMocks({tp, utils}, (mocks) => {
```
Comment (mykola-mokhnach) [inline test/general-specs.js]: actually idb requires minimum Python 3.6 to be installed

File: lib/ios.js [hunk 2]
```diff
@@ -146,26 +146,22 @@ checks.push(new CarthageCheck());
...
-        return okOptional(`fbsimctl is installed at: ${fbsimctlPath}. It is prbably installed as custom install.`);
-      }
+    const fbIdbPath = await resolveExecutablePath('idb');
+    const fbCompanionIdbPath = await resolveExecutablePath('idb_companion');
+    if (fbIdbPath && fbCompanionIdbPath) {
+      return okOptional('idb and idb_companion are installed');
     }
 
-    return nokOptional('fbsimctl cannot be found');
+    return nokOptional('idb and/or idb_companion are not installed');
```
Comment (mykola-mokhnach) [inline lib/ios.js]: I'd rather show separate error messages for idb and its companion

File: test/general-specs.js (hunk 1 above)
Comment (KazuCocoa) [reply test/general-specs.js]: yeah, good point.
In this case, it was enough if `idb` command existed as this doctor, I thought.

If a user has no `idb` (or python 3.6), we'll guide them to https://github.com/appium/appium-idb . Then the user can see the python requirement. If the user already has idb, we can expect the user has python 3 env.

File: lib/ios.js [hunk 3]
```diff
@@ -146,26 +146,28 @@ checks.push(new CarthageCheck());
...
+    const fbIdbPath = await resolveExecutablePath('idb');
+    const fbCompanionIdbPath = await resolveExecutablePath('idb_companion');
+    if (fbIdbPath && fbCompanionIdbPath) {
+      return okOptional('idb and idb_companion are installed');
     }
 
-    return nokOptional('fbsimctl cannot be found');
+    if (!fbIdbPath && fbCompanionIdbPath) {
+      return nokOptional('idb is not installed');
+    } else if (fbIdbPath && !fbCompanionIdbPath) {
```
Comment (mykola-mokhnach) [inline lib/ios.js]: else is redundant

**Pull request merged**